src/handlers/settlement-window-close.js

```diff
@@ -7,32 +7,53 @@ const handler = (router, routesContext) => {
     router.put('/settlement-window-close/:settlementWindowId', async (ctx, next) => {
         const { startDate: fromDateTime, endDate: toDateTime } = ctx.request.body;
 
-        const resp = await axios.post(
-            `${routesContext.config.centralSettlementsEndpoint}/settlementWindows/${ctx.params.settlementWindowId}`,
-            {
-                state: 'CLOSED',
-                reason: 'Finance portal user request',
-            },
-            {
-                validateStatus: null,
-            },
-        );
+        // if an external settlement service is provided use that
+        if (routesContext.config.externalSettlementsEndpoint) {
+            try {
+                const resp = await axios.post(`${routesContext.config.externalSettlementsEndpoint}`, {});
+                if (resp.status === 202) {
+                    ctx.response.status = 200;
+                } else {
+                    ctx.response.status = 500;
+                }
+            } catch (err) {
+                ctx.response.status = 500;
+            } finally {
+                // this delay is introduced in order to give the external settlement API enough time
+                // to close the window
+                // TODO revise the implementation of the external settlement API so it sends back an
+                //  acknowledgment somehow
+                await sleep(3000);
+                ctx.response.body = await handlerHelpers.getSettlementWindows(
+                    routesContext,
+                    fromDateTime,
+                    toDateTime,
+                );
+            }
+        } else { // use the default OSS settlement service
```

Comment:
@shashi165 - can you use the default as first (using ! as first if check and then have the external settlement service as else option) ?

Reply:
@elnyry-sam-k I assume this is to provide better backward compatibility going forward?